### ISSUE TITLE
Other Tools

### ISSUE THREAD

**Issue by MikeInnes** [denizyuret/Klutz.jl#1]
I don't think we should get into benchmarking every tool under the sun, but it'd be good to be fairly comprehensive in terms of the Julia ecosystem, and perhaps have one point of reference from outside. Options:

* Zygote (I'll work on this)
* ReverseDiff (at least for a subset of models where it's convenient)
* PyTorch (represents a good standard for perf outside of Julia)
* DyNet – a little faster than PyTorch but harder to make part of the build
* TensorFlow – unfortunately though it's quite hard to good GPU microbenchmarks for TF though.

**Comment by denizyuret** [comment]
I agree, but would add AutoGrad and Capstan to the AD list. We also need GPU lists don't we (for alloc and kernels). For outside comparisons PyTorch could be sufficient because there are benchmarks comparing DyNet, TF etc to PyTorch.
